DuckDuckGo/Bookmarks/Model/BookmarkManager.swift

```diff
@@ -25,10 +25,12 @@ import os.log
 protocol BookmarkManager: AnyObject {
 
     func isUrlBookmarked(url: URL) -> Bool
+    func isAnyUrlVariantBookmarked(url: URL) -> Bool
```

Comment:
Got a question. Because we still use the old methods:

func isUrlBookmarked(url: URL) -> Bool
func getBookmark(forUrl url: String) -> Bookmark?

Is it possible that in some scenarios we still return false/nil because the url doesn’t match 100%?

I'm wondering if we should have implemented the logic you added within those methods without adding new ones or we actually need to separate them.

Reply:
@alessandroboron, that's a good question! In most of the use cases, we want a method that returns a bookmark only when there is 100% match. The new method I added is necessary only to find a bookmark in case we don't need the exact match as it is for the bookmark icon in the address bar. That means, both methods are necessary plus the separation makes things clearer.